components/unit-cards/building-description-card.tsx

```diff
@@ -0,0 +1,182 @@
+import slash from "slash";
+import {
+  Accordion,
+  Card,
+  Divider,
+  Flex,
+  Grid,
+  Group,
+  Image,
+  Stack,
+  Text,
+  Title,
+} from "@mantine/core";
+import { UnitDescription, UnitDescriptionCard } from "./unit-description-card";
+import { UnitUpgrade } from "./unit-upgrade-card";
+import { ResourceValues, StatsCosts } from "./cost-card";
+import { BuildingIcon, BuildingType } from "../../src/coh3";
+
+type BuildingDescription = {
+  /** Locstring value. Found at `screen_name/locstring/value`. */
+  screen_name: string;
+  /** Locstring value. Found at `help_text/locstring/value`. */
+  help_text: string;
+  /** Locstring value. Found at `extra_text/locstring/value`. */
+  extra_text: string;
+  /** Locstring value. Found at `brief_text/locstring/value`. */
+  brief_text: string;
+  /** File path. Found at `icon_name`. */
+  icon_name: string;
+  /** File path. Found at `symbol_icon_name`. */
+  symbol_icon_name: string;
+};
+
+export type BuildingSchema = {
+  types: BuildingType[];
+  desc: BuildingDescription;
+  /** Extracted from `ebpextensions\\spawner_ext` within the building `ebps`. */
+  units: Array<{ desc: UnitDescription; time_cost: ResourceValues }>;
+  /** Extracted from `ebpextensions\\cost_ext` within the building `ebps`. */
+  time_cost: ResourceValues;
+  /**
+   * @todo Do we need this?
+   * Extracted from `ebpextensions\\ability_ext` within the building `ebps`.
+   */
+  abilities?: []; // Unused right now.
+  /** Extracted from `ebpextensions\\upgrade_ext` within the building `ebps`. */
+  upgrades: UnitUpgrade[];
+  health: { hitpoints: number };
+};
+
+/**
+ * @todo Shall we re-use the `unit_upgrade_card` instead? Only change is the
+ * icon side. In the meantime, use this in case of new UI requirements for this
+ * type of card.
+ */
+const BuildingCardHeader = (
+  desc: BuildingDescription,
+  cost: ResourceValues,
+  health: BuildingSchema["health"],
+) => (
+  <Grid columns={4}>
+    <Grid.Col sm={3}>
+      <Flex direction="row" align="center" gap={16}>
+        <Image
+          width={96}
+          height={96}
+          fit="contain"
+          src={`/icons/${slash(desc.icon_name)}.png`}
```

Comment:
Btw we will need to use some function for the `src` for the images, so it's centralized.